packages/adapter/src/lib/adapter/adapter.ts

```diff
@@ -2492,7 +2492,7 @@ export class AdapterClass extends EventEmitter {
     }
 
     // external signature
-    setTimeout(cb: TimeoutCallback, timeout: number, ...args: any[]): NodeJS.Timeout | void;
+    setTimeout(cb: TimeoutCallback, timeout: number, ...args: any[]): ioBroker.Timeout;
```

Comment:
```suggestion
    setTimeout(cb: TimeoutCallback, timeout: number, ...args: any[]): ioBroker.Timeout | undefined;
```
right? The internal signature can return `undefined`.

Reply:
Yes we can do so but then we should also allow undefined be passed to clear methods else it will be unhandy again.